File: 0021 - Merge Two Sorted Lists/ Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

const int ZERO = [](){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    return 0;
}();

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode fake(-1);
        ListNode* last = &fake;
        while (l1 != NULL && l2 != NULL) {
            if (l2->val > l1->val) {
                last->next = l1;
                last = l1;
                l1 = l1->next;
            } else {
                last->next = l2;
                last = l2;
                l2 = l2->next;
            }
        }
        while (l1 != NULL) {
            last->next = l1;
            last = l1;
            l1 = l1->next;
        }
        while (l2 != NULL) {
            last->next = l2;
            last = l2;
            l2 = l2->next;
        }
        return fake.next;
    }
};

File: 0021 - Merge Two Sorted Lists/Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        fake = last = ListNode(0)
        while l1 and l2:
            if l1.val > l2.val:
                last.next = l2
                last = l2
                l2 = l2.next
            else:
                last.next = l1
                last = l1
                l1 = l1.next
        
        while l1:
            last.next = l1
            last = l1
            l1 = l1.next
        
        while l2:
            last.next = l2
            last = l2
            l2 = l2.next
            
        return fake.next
